server/svix-server/src/v1/endpoints/attempt.rs

```diff
@@ -287,7 +287,7 @@ fn list_attempts_by_endpoint_or_message_filters(
         }
 
         if let Some(channel) = channel {
-            query = query.filter(Expr::cust_with_values("channels ?? ?", vec![channel]));
+            query = query.filter(Expr::cust_with_values("channels @> $1", [channel.jsonb()]));
```

Comment:
This is a little error-prone. `$1` is dependent on this particular fragment staying the first placeholder. Is there' no way to give SeaORM some kind of position-independent placeholder which it can replace with the correctly indexed one when it builds the query?

Reply:
It's a bit deceptive, but AFAICT the variable substitution happens "eagerly", so the $N placeholder only needs to match the number of values passed into `Expr::cust_with_values`.

I tested this real quick by just adding another variable and printing the resulting sql
```rust
        if let Some(channel) = channel {
            query = query.filter(Expr::cust_with_values(
                "channels @> $1",
                [channel.clone().jsonb()],
            ));
            query = query.filter(Expr::cust_with_values("channels @> $1", [channel.jsonb()]));
        }

        tracing::warn!(
            "XXX {}",
            sea_orm::QueryTrait::build(&query, sea_orm::DbBackend::Postgres).to_string()
        )
```

```
XXX SELECT "messageattempt"."id", "messageattempt"."created_at", "messageattempt"."msg_id", "messageattempt"."msg_dest_id", "messageattempt"."endp_id", "messageattempt"."url", "messageattempt"."status", "messageattempt"."response_status_code", "messageattempt"."response", "messageattempt"."ended_at", "messageattempt"."trigger_type" FROM "messageattempt" INNER JOIN "message" ON "message"."id" = "messageattempt"."msg_id" WHERE "messageattempt"."endp_id" = 'ep_2LMjmmJ10MYEbAu7rafZuMkmud8' AND channels @> E'\"obits\"' AND channels @> E'\"obits\"''
```

The last bit `AND channels @> E'\"obits\"' AND channels @> E'\"obits\"'` being what matters. 

On the flip side, if you try and match the `$N` to the number of params in the final query, the query builder actually panics.

So this
```rust
            query = query.filter(Expr::cust_with_values(
                "channels @> $1",
                [channel.clone().jsonb()],
            ));
            query = query.filter(Expr::cust_with_values("channels @> $2", [channel.jsonb()]));
```

panics with
```
panicked at 'index out of bounds: the len is 1 but the index is 1'
```

I'll add a comment explaining this since it caught me off guard too.

Reply:
Ahh, neat!